### PR TITLE
Fix Combobox filtering with custom itemToString

### PR DESCRIPTION
**Pull request by colinking**
Fixes: #215

If you supply an `itemToString` to an `Autocomplete` / `Combobox`, we will now correctly fuzzy filter on the string values.

| Before | After |
| ------ | ----- |
| image | image |

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
   "sideEffects": false,
   "scripts": {
     "test": "xo && ava",
-    "prepublishOnly": "rm -rf esm commonjs && yarn run build",
+    "prepare": "rm -rf esm commonjs && yarn run build",
```
Comment (Rowno) [inline package.json]: Why did you change this? `prepublishOnly` is an npm lifecycle hook that makes sure evergreen is rebuilt when people publish.

Comment (colinking) [reply package.json]: Yep, so is `prepare` :) https://docs.npmjs.com/misc/scripts#description

I wish `npm` had a separate object for lifecycle hooks (`lifecycle: { prepare: ... }`), so that this is less so black magic 😬 

But from the docs (`prepare` is a superset of `prepublishOnly`):

```
prepare: Run both BEFORE the package is packed and published, on local npm install without any arguments, and when installing git dependencies (See below). This is run AFTER prepublish, but BEFORE prepublishOnly.
prepublishOnly: Run BEFORE the package is prepared and packed, ONLY on npm publish. (See below.)
```

Admittedly, I forgot this was still here -- I was mainly using it for testing in another repo. But in general, this hook builds `evergreen` when you add it as a dependency, which allows you to reference a git branch (f.e. `"evergreen-ui": "segmentio/evergreen#colin/item-to-string-autocomplete",`). It's pretty useful for testing a branch like this out in a production setting.

However, not really sure if we want to include this change since it may slow down install times.

Comment (colinking) [reply package.json]: Reverting this change and dismissing the review so I can ship this :)

**Comment by colinking** [comment]
Merged master + double-checked that all storybook entries for `Autocomplete` and `Combobox` worked.
